File: src/main/webapp/resources/js/controllers/employeeController.js
gymApp.controller('EmployeeController', ['$http', '$scope', function($http, $scope){

    $scope.findAllEmployees = function() {
        $http.get('/api/employee/all').success(function (data) {
            $scope.employees = data;
        });
    };

    $scope.updateEmployee = function(employee) {
        $http.post('/api/employee/update/'+employee.employeeId, JSON.stringify(employee)).success(function () {
            $scope.findAllEmployees();
        });
    };
    $scope.createEmployee = function(employee) {
        $http.post('/api/employee/create', JSON.stringify(employee)).success(function () {
            $scope.clearAdd(employee);
            $scope.findAllEmployees();
        });
    };
    $scope.deleteEmployee = function(id) {
        $http.delete('/api/employee/delete/'+id).success(function () {
            $scope.findAllEmployees();
        });
    };

    $scope.clearAdd = function(employee){
        angular.element('#first-name').value = null;
        angular.element('#last-name').value = null;
        angular.element('#employee-number').value = null;
        angular.element('#phone-number').value = null;

        employee.lastName = null;
        employee.firstName = null;
        employee.employeeNumber = null;
        employee.phoneNumber = null;
    };

    $scope.showForm = false;
    $scope.findAllEmployees();
}]);

File: src/main/java/com/javaclass/gymapp/controller/EmployeeController.java
package com.javaclass.gymapp.controller;

import com.javaclass.gymapp.domain.Employee;
import com.javaclass.gymapp.service.EmployeeService;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Controller
@RequestMapping("/api/employee")
public class EmployeeController {

    @Autowired
    EmployeeService employeeService;

    @ResponseBody
    @RequestMapping(value = "/all", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public List<Employee> findAllEmployees(){
        return employeeService.findAll();
    }

    @RequestMapping(value = "/update/{id}", method = RequestMethod.POST)
    public ResponseEntity updateEmployee(@RequestBody String jsonEmployee){

        JSONObject object = new JSONObject(jsonEmployee);

        Employee employee = new Employee();
        employee.setEmployeeId(object.getLong("employeeId"));
        employee.setFirstName(object.getString("firstName"));
        employee.setLastName(object.getString("lastName"));
        employee.setEmployeeNumber(object.getString("employeeNumber"));
        try{
            employee.setPhoneNumber(object.getString("phoneNumber"));
        }catch (Exception e){

        }
        try {
            employeeService.updateEmployee(employee);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

    @RequestMapping(value = "/create", method = RequestMethod.POST)
    public ResponseEntity createEmployee(@RequestBody String jsonEmployee){

        JSONObject object = new JSONObject(jsonEmployee);

        Employee employee = new Employee();
        employee.setFirstName(object.getString("firstName"));
        employee.setLastName(object.getString("lastName"));
        employee.setEmployeeNumber(object.getString("employeeNumber"));
        try{
            employee.setPhoneNumber(object.getString("phoneNumber"));
        }catch (Exception e){

        }
        try {
            employeeService.createEmployee(employee);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

    @RequestMapping(value = "/delete/{id}", method = RequestMethod.DELETE)
    public ResponseEntity deleteEmployee(@PathVariable("id") Long id){

        try {
            employeeService.deleteEmployee(id);
            return new ResponseEntity(HttpStatus.OK);
        }catch (Exception e){
            System.out.println(e.getMessage());
            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }
}